client/src/main/scala/com/all4journey/webapp/util/Fragments.scala

```diff
@@ -0,0 +1,148 @@
+      a(href := "/multiformProfile/password")("Change Password")
+    )
+  )
+}
+
+object NavPills extends NavPills
+
+trait AddressForm extends AddressTypePickler {
+
+  @JSExport
+  val emptyAddress = Address("0", "0", None, State("NONE", "Choose a state"), "", HomeAddressType, "Home")
+
+  @JSExport
+  def load(address: Option[Address]) = div(id := "addressFieldsDiv", "addressUuid".attr := address.getOrElse(emptyAddress).id)(
+
+    div(id := "streetAddressDiv", cls := "form-group")(
+      label(cls := "col-lg-3 control-label")("Street Address:"),
+      div(cls := "col-lg-8")(
+        div(id := "streetAddressHelpBlock"),
+        input(id := "streetAddress", name := "streetAddress", cls := "form-control", `type` := "text", value := address.getOrElse(emptyAddress).street.getOrElse(""))
+      )
+    ),
+    div(id := "stateDiv", cls := "form-group")(
+      label(cls := "col-lg-3 control-label")("State:"),
+      div(cls := "col-lg-8")(
+        div(id := "stateHelpBlock"),
+        div(cls := "ui-select")(
+          select(id := "userState", name := "userState", cls := "form-control partOfStateList")(
+            option(value := address.getOrElse(emptyAddress).state.id, selected := "selected")(address.getOrElse(emptyAddress).state.description)
+          )
+        )
+      )
+    ),
+    div(id := "zipCodeDiv", cls := "form-group")(
+      label(cls := "col-lg-3 control-label")("Zip code:"),
+      div(cls := "col-lg-8")(
+        div(id := "zipCodeHelpBlock"),
+        input(id := "zipCode", name := "zipCode", cls := "form-control partOfZipCodeList", `type` := "text", value := address.getOrElse(emptyAddress).zipCode)
+      )
+    )
+  )
+
+  def setViolationPrompts(violations: Set[Violation]) = {
+    if (!violations.isEmpty) {
+      //                      val content = dom.document.getElementById("content")
+      //                      for (v <- failureList)
+      //                        content.appendChild(p(v.description).render)
+      violations.foreach(violation =>
+        violation.description.getOrElse("") match {
+          case street if street.contains("street") =>
+            HtmlHelper.showHelpBlock("#streetAddressDiv", "streetAddressHelpBlock", "invalid street address format")
+          case state if state.contains("state") =>
+            HtmlHelper.showHelpBlock("#stateDiv", "stateHelpBlock", "please choose a state")
+          case zipCode if zipCode.contains("zipCode") =>
+            HtmlHelper.showHelpBlock("#zipCodeDiv", "zipCodeHelpBlock", "invalid zip code format")
+        }
+      )
+    }
+  }
+
+  def doValidation(address: Address): Set[Violation] = {
+    //val violations = Set.empty[Violation]
+    val fullAddressValidationResult = validate(address)(Address.validatorWithStreet)
+
+    if (fullAddressValidationResult.isFailure) {
+      val partialAddressValidationResult = if (address.street.getOrElse("").isEmpty && !address.state.id.equals("NONE"))
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Consider using case matching instead of else if blocks](https://www.codacy.com/app/rjkernick/all4journey/file/2510792589/issues/source?bid=3246460&fileBranchId=3273764#l90)